substrate/primitives/version/src/lib.rs

```diff
@@ -230,9 +221,406 @@ pub struct RuntimeVersion {
 	/// This number should never decrease.
 	pub transaction_version: u32,
 
-	/// Version of the state implementation used by this runtime.
+	/// Version of the system implementation used by this runtime.
 	/// Use of an incorrect version is consensus breaking.
-	pub state_version: u8,
+	pub system_version: u8,
+}
+
+// Manual implementation in order to sprinkle `stateVersion` at the end for migration purposes
+// after the field was renamed from `state_version` to `system_version`
+#[cfg(feature = "serde")]
+impl serde::Serialize for RuntimeVersion {
```

Comment:
Can we move all this code into a separate file? 400 lines is a lot.

Reply:
Would `serde` module work? I generally try to keep impls next to the definition, but agree that in this case it is quite lengthy.